### PR TITLE
[circle-quantizer] Support quantization of softmax

### PR DESCRIPTION
**Pull request by jinevening**
This supports quantization of softmax op.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: #6367

### CODE REVIEW

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 1]
```diff
@@ -199,6 +219,45 @@ TEST(QuantizedModelVerifierTest, Logistic_wrong_granularity_NEG)
...
+TEST(QuantizedModelVerifierTest, Softmax)
+{
+  TEST_WITH_GRAPH(SoftmaxTestGraph, Type::U8, Granularity::LayerWise);
+  TEST_WITH_GRAPH(SoftmaxTestGraph, Type::U8, Granularity::ChannelWise);
+  TEST_WITH_GRAPH(SoftmaxTestGraph, Type::S16, Granularity::ChannelWise);
+}
+
+TEST(QuantizedModelVerifierTest, Softmax_wrong_type_NEG)
+{
+  {
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:231]: This brace seems needless :)

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 2]
```diff
@@ -199,6 +219,45 @@ TEST(QuantizedModelVerifierTest, Logistic_wrong_granularity_NEG)
...
+    g._softmax->dtype(Type::S16);
+
+    luci::QuantizedModelVerifier verifier(Type::U8, Granularity::LayerWise);
+    EXPECT_ANY_THROW(verifier.verify(g.g()));
+  }
+}
+
+TEST(QuantizedModelVerifierTest, Softmax_wrong_granularity_NEG)
+{
+  {
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:247]: this too

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp (hunk 1 above)
Comment (jinevening) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:231]: It is for avoiding name aliasing when adding a new test in the same category (testing wrong type).

Without the brackets, we'll have to use different variable names when adding a new test, for example,

```cpp
// test 1
SoftmaxTestGraph g;
..

// test 2
SoftmaxTestGraph g2;
...
```

Comment (seanshpark) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:231]: (optional) For this case, IMHO, adding one test for each in test case would be better, that is, 
something like, `Softmax_wrong_type_NEG` -> `Softmax_wrong_type_LW_U8_NEG`
I'll wait for your decision :)

Comment (jinevening) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp]: Dividing the test case looks good. I've updated the code. PTAL

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**